centrosome/cpmorphology.py

```diff
@@ -464,9 +464,9 @@ def strel_line(length, angle):
     Note: uses draw_line's Bresenham algorithm to select points.
     """
     angle = float(angle) * np.pi / 180.0
-    x_off = int(np.finfo(float).eps + np.cos(angle) * length / 2)
+    x_off = int(np.round(np.finfo(float).eps + np.cos(angle) * length / 2))
```

Comment:
Consider using `scipy.special.cosdg()` and `sindg()` to work directly with degrees. I assume the tricks with `eps` and `round()` are to make sure that fun special cases like `angle = 90.0` get clean `0`s. `cosdg()` and `sindg()` help with that.

Reply:
Thanks! I will take a look, but at least from local testing on my Mac, it still seems like we may still need to keep one or both of the eps and round tricks: the scipy functions help with 0s but not ie 0.5
```
>>> import numpy as np
>>> import scipy.special
>>> fn_dict = {np.sin:scipy.special.sindg,np.cos:scipy.special.cosdg,np.tan:scipy.special.tandg}
>>> for fn in fn_dict.keys():
...     for angle in range(0,181,30):
...             print(str(fn),angle,fn(angle*np.pi/180),fn_dict[fn](angle))
... 
<ufunc 'sin'> 0 0.0 0.0
<ufunc 'sin'> 30 0.49999999999999994 0.49999999999999994
<ufunc 'sin'> 60 0.8660254037844386 0.8660254037844387
<ufunc 'sin'> 90 1.0 1.0
<ufunc 'sin'> 120 0.8660254037844388 0.8660254037844387
<ufunc 'sin'> 150 0.49999999999999994 0.49999999999999994
<ufunc 'sin'> 180 1.2246467991473532e-16 -0.0
<ufunc 'cos'> 0 1.0 1.0
<ufunc 'cos'> 30 0.8660254037844387 0.8660254037844387
<ufunc 'cos'> 60 0.5000000000000001 0.49999999999999994
<ufunc 'cos'> 90 6.123233995736766e-17 -0.0
<ufunc 'cos'> 120 -0.4999999999999998 -0.49999999999999994
<ufunc 'cos'> 150 -0.8660254037844387 -0.8660254037844387
<ufunc 'cos'> 180 -1.0 -1.0
<ufunc 'tan'> 0 0.0 0.0
<ufunc 'tan'> 30 0.5773502691896256 0.5773502691896256
<ufunc 'tan'> 60 1.7320508075688767 1.7320508075688767
<ufunc 'tan'> 90 1.633123935319537e+16 inf
<ufunc 'tan'> 120 -1.7320508075688783 -1.7320508075688767
<ufunc 'tan'> 150 -0.5773502691896256 -0.5773502691896256
<ufunc 'tan'> 180 -1.2246467991473532e-16 0.0
```